lib/Common/Common/Int64Math.cpp

```diff
@@ -5,8 +5,23 @@
 #include "CommonCommonPch.h"
 #include "Common/Int64Math.h"
 
-#if defined(_M_X64) && defined(_MSC_VER)
-#pragma intrinsic(_mul128)
+#if defined(_M_X64)
+#if defined(_MSC_VER)
+    #pragma intrinsic(_mul128)
+#else
+    static int64 _mul128(const int64 left, const int64 right, int64 *high) noexcept
+    {
+        int64 low;
+        __asm__
+        (
+            "imulq %[right]\n"
+            : "=d"(*high), "=a"(low)
+            : [right]"rm"(right), "1"(left)
```

Comment:
I don't know much about gcc's inline asm.  
Looking at the doc, "1"(left) means that you require left to be at the same location as output 1 (which rax).
Am I right?  Why?